### PR TITLE
added AD Penalty Dirichlet BCs

### PR DESCRIPTION
**Pull request by tophmatthews**
Added ADPenaltyDirichletBC and ADFunctionDirichletBC, and matched to non-AD methods.

Ref #12903

### CODE REVIEW

**Comment by tophmatthews** [comment]
@permcody is this you canceling these jobs? I have some for BISON that were canceled yesterday too without explanation. Just let me know how to better manage testing if I am doing it wrong.

**Comment by tophmatthews** [comment]
Maybe @lindsayad or @dschwen ?

File: framework/doc/content/source/bcs/ADPenaltyDirichletBC.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# ADPenaltyDirichletBC
+
+!syntax description /BCs/ADPenaltyDirichletBC
+
+## Description
+
+`ADPenaltyDirichletBC` is a `ADNodalBC` used for enforcing Dirichlet boundary conditions
```
Comment (lindsayad) [inline framework/doc/content/source/bcs/ADPenaltyDirichletBC.md]: `Nodal` -> `Integrated`. You scared me for a second

File: test/tests/bcs/ad_penalty_dirichlet_bc/tests [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      detail = 'with a constant value on the boundary and shall be able to produce the exact Jacobian.'
+    [../]
+
+    [./function_penalty_dirichlet_bc_test]
+      type = 'Exodiff'
+      input = 'function_penalty_dirichlet_bc_test.i'
+      exodiff = 'function_penalty_dirichlet_bc_test_out.e'
+      abs_zero = 1e-05
+      max_parallel = 11
+      scale_refine = 3
```
Comment (lindsayad) [inline test/tests/bcs/ad_penalty_dirichlet_bc/tests]: Why so refined?

**Review by lindsayad — CHANGES_REQUESTED** [review]
LGTM other than the one documentation fix

File: framework/doc/content/source/bcs/ADPenaltyDirichletBC.md (hunk 1 above)
Comment (tophmatthews) [reply framework/doc/content/source/bcs/ADPenaltyDirichletBC.md]: whoops, nice. This is just copied from the non-AD text. I'll update both.

File: test/tests/bcs/ad_penalty_dirichlet_bc/tests (hunk 2 above)
Comment (tophmatthews) [reply test/tests/bcs/ad_penalty_dirichlet_bc/tests]: This just duplicates the non-AD tests...good question. nx/ny=10 already in the input file. Should I drop it or leave it?

Comment (lindsayad) [reply test/tests/bcs/ad_penalty_dirichlet_bc/tests]: This option may not actually do anything any more IIRC. @permcody ?

**Review by lindsayad — APPROVED** [review]

**Comment by permcody** [comment]
@tophmatthews - It appears that your test is just outside of tolerance in parallel. Can you take a look?
https://civet.inl.gov/branch/6695/

**Pull request merged**